Client/Frontend/Browser/BrowserLongPressGestureRecognizer.swift

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import UIKit
+
+enum ElementType {
+    case ImageSrc(NSURL)
+    case hrefLink(NSURL)
+}
+
+protocol BrowserLongPressGestureDelegate: class {
+    func browser(browser: Browser, elementsDidLongPressed elements: [ElementType])
```

Comment:
For consistency, please rename <code>browser</code> -> <code>longPressRecognizer</code>, and pass a <code>LongPressGestureRecognizer</code> instead of a <code>Browser</code>. Also, with the dictionary changes above, the <code>elements</code> type should change to a <code>[LongPressElementType: NSURL]</code> dictionary.

So, in summary:
```
func longPressRecognizer(longPressRecognizer: LongPressGestureRecognizer, 
    didLongPressElements elements: [LongPressElementType: NSURL])
```